### ISSUE TITLE
SingleChoiceAnswer sometimes does not return value

### ISSUE THREAD

**Issue by bobhuang1** [bcemmett/SurveyMonkeyApi-v3#12]
Need help! I have a survey has a "country" filed, which is a single choice menu with all countries listed (long list but very simple). Somehow when the user selects "United States" as the value, latest API does not return the correct value for Choice.

Code:
```csharp
using (var api = new SurveyMonkeyApi("myKey", "myToken"))
{
    Survey survey = api.PopulateSurveyResponseInformation(mySurveyID);
    Response.Write("Total response count: "+ survey.Responses.Count +"<br />");
    foreach (Response myResponse in survey.Responses)
    {
        foreach (ResponseQuestion myResponseQuestion in myResponse.Questions)
        {
            Response.Write(myResponseQuestion.ProcessedAnswer.Response + " " + myResponseQuestion.ProcessedAnswer.QuestionFamily + " " + myResponseQuestion.ProcessedAnswer.QuestionSubtype + "<br />");

            if (myResponseQuestion.ProcessedAnswer.QuestionFamily == QuestionFamily.SingleChoice)
            {
                Response.Write(((SingleChoiceAnswer)myResponseQuestion.ProcessedAnswer.Response).Choice + "<br />");
                // This is where it fails to return United States, however it does return other countries, weird!
            }
        }
    }
}
```
Output from code:
SurveyMonkey.ProcessedAnswers.SingleChoiceAnswer SingleChoice Menu
-(this is where the country is supposed to show up, but it's empty)

**Comment by bcemmett** [comment]
That sounds very strange. If you run fiddler, can you see whether "United States" is being correctly returned by the api? I can't imagine a reason that the library would treat that value any differently if it was being returned.

**Comment by bobhuang1** [comment]
Yes, "United States" is being correctly returned from survey monkey. But now this problem has gone worse. For some countries like "United Kingdom", it also returns an empty string.

**Comment by bcemmett** [comment]
So to clarify, The json coming back from surveymonkey says "United Kingdom", but the object value this library gives is an empty string?

**Comment by bcemmett** [comment]
Any chance you can post the complete json?

**Comment by bobhuang1** [comment]
Actually I have found the cause of this, but I do not know where to fix. I am using PopulateSurveyResponseInformation function. It's supposed to retrieve everything including questions and processed answers. But when it comes down to the country field, 

using (var api = new SurveyMonkeyApi("myKey", "myToken"))
{
    Survey survey = api.PopulateSurveyResponseInformation(mySurveyID);
}

Data in survey -> Pages -> [0] - > Questions -> [1] (my country question -> Answers -> Choices *****
Under choices the counter is 100, which means it's limited to the fist 100 countries returned, and that's the cause since countries started with "united" are far down in the list, so they are not matched and thus a null is returned!

**Comment by bobhuang1** [comment]
In summary, since that question's answer contains more than 100 choices, the answers above 100 are cut off, and thus not able to find a match. I suspect it has something to do with the maxResultsPerPage in SurveyMonkeyApi.Surveys.cs  and SurveyMonkeyApi.Responses.cs. Thanks.

**Comment by bcemmett** [comment]
When you look at json which comes back from the api for the request made against "/surveys/{SurveyId}/details", does that include all the >100 choices and the library is somehow removing them after 100 (can't see how that would happen at the moment), or are they omitted by the api?

**Comment by bobhuang1** [comment]
The json returns only fist 100. And anything after are removed.

**Comment by bcemmett** [comment]
Ah, I'm afraid if it's the api which is truncating the data then that sounds like a bug in the api itself rather than the library. Suggest you contact their support team at api-support@surveymonkey.com to see if they can help you out. Let me know if they fix it!

**Comment by bobhuang1** [comment]
Thank you, Ben. You are right! I sent an Email to them and I just got a reply from the Survey Monkey support team. They agree it's a bug in their API and will fix it ASAP. In the meantime, I have implemented a fix by using the /surveys/{id}/pages/{id}/questions/{id} endpoint directly - it does return all choices even though there 197 of them. Their user portal does turn correct data while the API /surveys/{id}/details does not! That got me confused. But thanks again for the prompt replies, Ben!
